### PR TITLE
Port Switch

### LINKED ISSUE #9: Enable/Disable Port forwardings
Evaluate whether that's possible.

Nice use-case: Enable port 80 port forwarding to allow the let's encrypt http challenge to pass.

### PR DESCRIPTION
**Pull request by AaronDavidSchneider**
I tried implementing the port switches in the same manner as the guest wifi switch. I hope it works that way.
Not working yet:
1. ipv6 toggle of ports (probably will never)
2. automatic discovery of devices ip. Till now: only  manual setting in configuration.yaml
3. entity_id not set, only names set.
4. Opened ports are only discovered when component is loaded

Debugging probably still necessary

Fixes #9

### CODE REVIEW

**Comment by AaronDavidSchneider** [comment]
I finished my work on the port switch and it works!

Todos for the switch would be:
1. ipv6 support (check if that's possible at all)
2. automatic discovery of devices ip.

**Comment by mammuth** [comment]
Cool, thanks! :tada: 
I try my best to review and integrate it during the next days!

File: custom_components/fritzbox_tools/__init__.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ def setup(hass, config):
     port = config[DOMAIN].get('port', 49000)
     username = config[DOMAIN].get('username', '')
     password = config[DOMAIN].get('password', None)
+    ip_device = config[DOMAIN].get('ip_device', "192.168.178.4")
```
Comment (mammuth) [inline custom_components/fritzbox_tools/__init__.py]: That default doesn't make much sense I think. I'll replace it with `None` afterward.

File: custom_components/fritzbox_tools/switch.py [hunk 2]
```diff
@@ -16,17 +16,117 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     _LOGGER.debug('Setting up switches')
     fritzbox_tools = hass.data[DOMAIN][DATA_FRITZ_TOOLS_INSTANCE]
     # add_entities([FritzBoxGuestWifiSwitch(fritzbox_tools)])
-    add_entities([FritzBoxGuestWifiSwitch(fritzbox_tools)], True)
+    n_ports_total = fritzbox_tools.connection.call_action('WANIPConnection:1', 'GetPortMappingNumberOfEntries')["NewPortMappingNumberOfEntries"]
```
Comment (mammuth) [inline custom_components/fritzbox_tools/switch.py]: Nice find. Much better than my initial thought of randomly iterating over eg. `range(30)` 🙈 😁

**Comment by mammuth** [comment]
I'm able to see my port switches (for the device running HA) in HA. However, turning on/off the switches doesn't change the port forwardings on my router. I'll try to further debug it.

**Comment by AaronDavidSchneider** [comment]
It doesn’t? do you have „Selbstständige portfreigabe“ enabled? It did enable and disable on my Fritzbox

**Comment by mammuth** [comment]
> do you have „Selbstständige portfreigabe“ enabled?
Nope, didn't 🙈 It works fine after doing so.

I think I'll merge it now and do some small adjustments afterward! Thanks for your great contribution! 🎉

**Comment by mammuth** [comment]
Btw, since you said you're inexperienced with the git workflow:
It makes sense to create a new branch in your fork before creating a pull request. This way it's easier to work on multiple things in parallel (eg. pushing hotfixes or stuff like that) 😉

**Comment by AaronDavidSchneider** [comment]
Thanks. I know, I use branches as well. Wasn’t really needed here though.

**Pull request merged**